File: hackfiniti/notibuzz/frontend/src/types/email.ts
export type PriorityLevel = 'urgent' | 'important' | 'normal' | 'low'

export type EmailCategory = 'work' | 'personal' | 'financial' | 'marketing' | 'social' | 'newsletter' | 'other'

export interface EmailAttachment {
  filename: string
  size: number
  content_type: string
  attachment_id?: string
}

export interface Email {
  id: string
  thread_id: string
  subject: string
  sender: string
  sender_email: string
  recipients: string[]
  cc: string[]
  bcc: string[]
  body_text: string
  body_html?: string
  timestamp: string
  read: boolean
  starred: boolean
  important: boolean
  attachments: EmailAttachment[]
  labels: string[]
  
  // AI-generated fields
  summary?: string
  priority: PriorityLevel
  category: EmailCategory
  sentiment_score?: number
  embedding?: number[]
  
  // Processing metadata
  processed_at?: string
  last_updated: string
}

export interface EmailSearch {
  query: string
  limit: number
  offset: number
  filters?: Record<string, any>
}

export interface EmailSearchResult {
  emails: Email[]
  total_count: number
  query: string
  processing_time: number
}

export interface EmailAnalytics {
  total_emails: number
  unread_count: number
  urgent_count: number
  important_count: number
  category_breakdown: Record<string, number>
  daily_volume: Record<string, number>
  top_senders: Record<string, number>
}

File: hackfiniti/notibuzz/backend/app/models/email.py
from pydantic import BaseModel, Field
from typing import Optional, List, Dict, Any
from datetime import datetime
from enum import Enum


class PriorityLevel(str, Enum):
    URGENT = "urgent"
    IMPORTANT = "important"
    NORMAL = "normal"
    LOW = "low"


class EmailCategory(str, Enum):
    WORK = "work"
    PERSONAL = "personal"
    FINANCIAL = "financial"
    MARKETING = "marketing"
    SOCIAL = "social"
    NEWSLETTER = "newsletter"
    OTHER = "other"


class EmailAttachment(BaseModel):
    filename: str
    size: int
    content_type: str
    attachment_id: Optional[str] = None


class Email(BaseModel):
    id: str
    thread_id: str
    subject: str
    sender: str
    sender_email: str
    recipients: List[str]
    cc: List[str] = []
    bcc: List[str] = []
    body_text: str
    body_html: Optional[str] = None
    timestamp: datetime
    read: bool = False
    starred: bool = False
    important: bool = False
    attachments: List[EmailAttachment] = []
    labels: List[str] = []
    
    # AI-generated fields
    summary: Optional[str] = None
    priority: PriorityLevel = PriorityLevel.NORMAL
    category: EmailCategory = EmailCategory.OTHER
    sentiment_score: Optional[float] = None
    embedding: Optional[List[float]] = None
    
    # Processing metadata
    processed_at: Optional[datetime] = None
    last_updated: datetime = Field(default_factory=datetime.utcnow)


class EmailSearch(BaseModel):
    query: str
    limit: int = Field(default=20, ge=1, le=100)
    offset: int = Field(default=0, ge=0)
    filters: Optional[Dict[str, Any]] = None


class EmailSearchResult(BaseModel):
    emails: List[Email]
    total_count: int
    query: str
    processing_time: float


class EmailAnalytics(BaseModel):
    total_emails: int
    unread_count: int
    urgent_count: int
    important_count: int
    category_breakdown: Dict[str, int]
    daily_volume: List[Dict[str, Any]]
    top_senders: List[Dict[str, Any]]
